Terminal.Gui/Views/Scroll/ScrollButton.cs

```diff
@@ -0,0 +1,142 @@
+﻿#nullable enable
+
+namespace Terminal.Gui;
+
+internal enum VariationMode
+{
+    Decrease,
+    Increase
+}
+
+internal class ScrollButton : View
+{
+    public ScrollButton ()
+    {
+        TextAlignment = Alignment.Center;
+        VerticalTextAlignment = Alignment.Center;
+        Id = "scrollButton";
+
+        //Width = Dim.Auto (DimAutoStyle.Content, 1);
+        //Height = Dim.Auto (DimAutoStyle.Content, 1);
+        WantContinuousButtonPressed = true;
+    }
+
+    private ColorScheme? _savedColorScheme;
+
+    public void AdjustButton ()
+    {
+        if (!IsInitialized)
+        {
+            return;
+        }
+
+        Width = SupView.Orientation == Orientation.Vertical ? Dim.Fill () : 1;
+        Height = SupView.Orientation == Orientation.Vertical ? 1 : Dim.Fill ();
+
+        switch (VariationMode)
+        {
+            case VariationMode.Decrease:
+                X = 0;
+                Y = 0;
+
+                break;
+            case VariationMode.Increase:
+                X = SupView.Orientation == Orientation.Vertical ? 0 : Pos.AnchorEnd (1);
+                Y = SupView.Orientation == Orientation.Vertical ? Pos.AnchorEnd (1) : 0;
+
+                break;
+            default:
+                throw new ArgumentOutOfRangeException ();
+        }
+
+        SetButtonText ();
+    }
+
+    /// <inheritdoc/>
+    public override Attribute GetNormalColor ()
+    {
+        if (_savedColorScheme is null)
+        {
+            ColorScheme = new () { Normal = new (SupView.ColorScheme.HotNormal.Foreground, SupView.ColorScheme.HotNormal.Background) };
+        }
+        else
+        {
+            ColorScheme = new () { Normal = new (SupView.ColorScheme.Normal.Background, SupView.ColorScheme.Normal.Foreground) };
+        }
+
+        return base.GetNormalColor ();
+    }
+
+    public VariationMode VariationMode { get; init; }
```

Comment:
`internal`

Reply:
The `ScrollButton` class and the `VariationMode` enum are already internal and thus the public keyword is limited by the internal keyword of their main objects.

Reply:
I know that. For correctness tho...